### PR TITLE
Update docsite generated materials

### PR DESCRIPTION
**Pull request by kaidokert**
Updates Docsite for removed Starboard 12 and adds Starboard 15, along with out doc tweaks.

b/286653238

### CODE REVIEW

**Comment by kaidokert** [comment]
Internal chain go/cobalt-cl/252484

File: cobalt/site/docs/gen/cobalt/doc/device_authentication.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ function to sign the message using the secret key.  This method is preferred
 since it enables implementations where the key exists only in secure hardware
 and never enters the system's main memory.  A reference implementation, which
 depends on BoringSSL exists at
-[starboard/linux/x64x11/internal/system_sign_with_certification_secret_key.cc](../../starboard/linux/x64x11/internal/system_sign_with_certification_secret_key.cc).
+[internal/starboard/linux/x64x11/internal/system_sign_with_certification_secret_key.cc](../../internal/starboard/linux/x64x11/internal/system_sign_with_certification_secret_key.cc).
```
Comment (andrewsavage1) [inline cobalt/site/docs/gen/cobalt/doc/device_authentication.md:42]: Let's change this to a code snipped

Comment (kaidokert) [reply cobalt/site/docs/gen/cobalt/doc/device_authentication.md]: Done. I don't know why we are editing the same .md doc in 2 different places now but oh well.

File: cobalt/site/docs/gen/cobalt/doc/docker_build.md [hunk 2]
```diff
@@ -67,4 +79,4 @@ launching the bash shell, i.e.
 
   `docker-compose run linux-x64x11 /bin/bash`
 
-and try to build cobalt [with the usual `gyp / ninja` flow](../../README.md#building-and-running-the-code).
+and try to build cobalt [with the usual `gn / ninja` flow](../../README.md#building-and-running-the-code).
```
Comment (oxve) [inline cobalt/site/docs/gen/cobalt/doc/docker_build.md:82]: 😬

Comment (kaidokert) [reply cobalt/site/docs/gen/cobalt/doc/docker_build.md:82]: Hey look, i fixed it. There's quite few other gyp mentions in the docs but i don't need to win them all right away.

**Review by oxve — APPROVED** [review]

**Pull request merged**